samples/ControlCatalog/Pages/TreeViewPage.xaml.cs

```diff
@@ -1,40 +1,115 @@
-using System.Collections;
-using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Linq;
+using System.Reactive;
 using Avalonia.Controls;
 using Avalonia.Markup.Xaml;
+using ReactiveUI;
 
 namespace ControlCatalog.Pages
 {
     public class TreeViewPage : UserControl
     {
         public TreeViewPage()
         {
-            this.InitializeComponent();
-            DataContext = new Node().Children;
+            InitializeComponent();
+            DataContext = new PageViewModel(this.Find<TreeView>("treeView"));
         }
 
         private void InitializeComponent()
         {
             AvaloniaXamlLoader.Load(this);
         }
 
-        public class Node
+        private class PageViewModel : ReactiveObject
         {
-            private IList<Node> _children;
+            private readonly TreeView _treeView;
+            private SelectionMode _selectionMode;
+
+            public PageViewModel(TreeView treeView)
```

Comment:
I don't like that you pass a control to your viewmodel. Better bind SelectedItems and SelectedItem to a property in your viewmodel and avoid using the TreeView directly that way.